bittensor/axon.py

```diff
@@ -151,6 +151,61 @@ def stop(self):
             self.should_exit = True
 
 
+def load_nonces(
+    nonces_basepath: str, coldkey: str, hotkey: str, ip: str, port: int
+) -> Dict[str, int]:
+    """
+    Loads the nonces from the specified file path using hash of coldkey:hotkey:netuid and returns them as a dictionary.
+
+    Args:
+        nonces_path (str): The file path to load the nonces from.
+
+    Returns:
+        Dict[str, int]: A dictionary containing the (endpoint_key, nonce) pairs loaded from the file.
+    """
+    if not os.path.exists(nonces_basepath):
+        os.makedirs(nonces_basepath)
+
+    nonce_dirname = str(hash(f"{coldkey}:{hotkey}:{ip}:{port}"))
+    nonces_path = os.path.join(nonces_basepath, nonce_dirname, "nonces.json")
+
+    if os.path.exists(nonces_path):
+        with open(nonces_path, "r") as f:
+            nonces = json.load(f)
+        return nonces
+
+    return {}
```

Comment:
```suggestion
    nonce_dirname = str(hash(f"{coldkey}:{hotkey}:{ip}:{port}"))
    nonces_path = os.path.join(nonces_basepath, nonce_dirname, "nonces.json")

    try:
        with open(nonces_path, "r") as f:
            nonces = json.load(f)
        return nonces
    except (FileNotFoundError, json.JSONDecodeError):
        return {}
```
other than less code, the important difference is that this one doesn't contain race condition if file was deleted for some reason.  The second important difference is that, if file was not completed (due error during write, or just because write operation in other process has yet to complete), it will not error out.

I think it would be best to add read/write lock here if we wanted this to be rock solid, but I guess this will do.